packages/react/menu/src/Menu.tsx

```diff
@@ -189,15 +194,39 @@ const MenuImpl = forwardRefWithAs<typeof PopperPrimitive.Root, MenuImplOwnProps>
       <PortalWrapper>
         <ScrollLockWrapper>
           <FocusScope
-            trapped={trapFocus}
+            // clicking outside may raise a focusout event, which may get trapped.
+            // in cases where outside pointer events are permitted, we stop trapping.
+            trapped={isPermittedPointerDownOutsideEvent ? false : trapFocus}
             onMountAutoFocus={onOpenAutoFocus}
-            onUnmountAutoFocus={onCloseAutoFocus}
+            onUnmountAutoFocus={(event) => {
+              // skip autofocus on unmount if clicking outside is permitted and it happened
+              if (isPermittedPointerDownOutsideEvent) {
+                event.preventDefault();
+              } else {
+                onCloseAutoFocus?.(event);
+              }
+            }}
           >
             {(focusScopeProps) => (
               <DismissableLayer
                 disableOutsidePointerEvents={disableOutsidePointerEvents}
                 onEscapeKeyDown={onEscapeKeyDown}
-                onPointerDownOutside={onPointerDownOutside}
+                onPointerDownOutside={composeEventHandlers(
+                  onPointerDownOutside,
+                  (event) => {
+                    const isNotRightClick =
```

Comment:
`const isLeftClick` ? I always struggle with `isNot` because we have `!is` for that.

Also, if we type the event above this line will that prevent your wrapping here? 🙈 haha

```jsx
(event: MouseEvent) => {
	const isLeftClick = event.button === 0 && event.ctrlKey === false;
	...
})
```

Reply:
haha I know, same here!

Except in this case, it can be a `TouchEvent` and `event.button` doesn't exist on a `TouchEvent` (that's why I'm casting as `MouseEvent`).
So saying `isLeftClick` would not really be correct.
Whereas calling it `isNotRightClick` is actually correct in all cases.

Can you think of another way?